src/dodal/devices/i10/id_apple2.py

```diff
@@ -0,0 +1,464 @@
+    """
+
+    def __init__(
+        self,
+        id_gap: UndulatorGap,
+        id_phase: UndlatorPhaseAxes,
+        energy_gap_table_path: Path,
+        energy_phase_table_path: Path,
+        source: tuple[str, str] | None = None,
+        prefix: str = "",
+        mode: str = "Mode",
+        min_energy: str = "MinEnergy",
+        max_energy: str = "MaxEnergy",
+        poly_deg: list | None = None,
+        name: str = "",
+    ) -> None:
+        super().__init__(name)
+
+        # Attributes are set after super call so they are not renamed to
+        # <name>-undulator, etc.
+        with self.add_children_as_readables():
+            self.gap = id_gap
+            self.phase = id_phase
+        with self.add_children_as_readables(HintedSignal):
+            # Store the polarisation for readback.
+            self.polarisation, self._polarisation_set = soft_signal_r_and_setter(
+                str, initial_value=None
+            )
+            # Store the set energy for readback.
+            self.energy, self._energy_set = soft_signal_r_and_setter(
+                float, initial_value=None
+            )
+
+        # A dictionary contains the path to the look up table and the expected column names.
+        self.lookup_table_config = {
+            "path": {
+                "Gap": energy_gap_table_path,
+                "Phase": energy_phase_table_path,
+            },
+            "source": source,
+            "mode": mode,
+            "min_energy": min_energy,
+            "max_energy": max_energy,
+            "poly_deg": poly_deg,
+        }
+        self.lookup_tables = {}
+        self._available_pol = []
+        self._pol = None
+        # Run at start up to load lookup tables and set available_pol.
+        self.update_poly()
+
+    @property
+    def pol(self):
+        return self._pol
+
+    @pol.setter
+    def pol(self, pol: str):
+        # This set the pol but does not actually move hardware.
+        if pol in self._available_pol:
+            self._pol = pol
+        else:
+            raise ValueError(
+                f"Polarisation {pol} is not available:"
+                + f"/n Polarisations available:  {self._available_pol}"
+            )
+
+    @AsyncStatus.wrap
+    async def set(self, value: float) -> None:
+        """
+        Check polarisation state and use it together with the energy(value)
+        to calculate the required gap and phases before setting it.
+        """
+        if self.pol is None:
+            LOGGER.warning("Polarisation not set attempting to read from hardware")
+            pol, phase = await self.determinePhaseFromHardware()
+            if pol is None:
+                raise ValueError(f"Pol is not set for {self.name}")
+            else:
+                self.pol = pol
+
+        self._polarisation_set(self.pol)
+
+        gap, phase = self._get_id_gap_phase(value)
+        phase3 = phase if self.pol != "la" else -phase
```

Comment:
```suggestion
        phase3 = -phase if self.pol == "la" else phase
```

Reply:
As "la" is the rare case I prefer the original, as phase does not have to change so I save the -1 operation :p